### ISSUE TITLE
Replay browser credentials should not leak to page

### ISSUE THREAD

**Issue by kannanvijayan** [RecordReplay/gecko-dev#414]
@jasonLaster @bhackett1024

Not sure if there exists an issue for this - did a quick search and did not find any.

Currently, logging in to replay in the replay browser with google credentials is equivalent to logging into google within the browser and storing auth credentials the way a regular browser would store them - i.e. credentials that are available to pages that are visited in the browser.

This unnerves and disturbs me every time I run into it, and is a security issue IMHO.  It also directly works against our goal of encouraging people to share their recordings.  From a user's perspective, any webpage with an embedded google auth widget (visible or invisible) leaks info about them.  This may be true if you're recording gdocs, or if you're recording cnn.com.  Basically users cannot reliably share recordings of any third party sites if this behaviour stays the way it is.

The replay browser should store the replay user's login credentials in a separate location that is not accessible to pages (a pref containing the auth token?).

TL;DR:  Authenticating yourself to the replay browser with a google account MUST NOT have the same behaviour as authentication yourself to google with the replay browser.

Thoughts?
